src/react-apps/applications/runtime/src/shared/resources/queue/queueReducer.ts

```diff
@@ -0,0 +1,33 @@
+import update from 'immutability-helper';
+import { Action, Reducer } from 'redux';
+import * as DataTaskQueueActionTypes from './dataTask/dataTaskQueueActionTypes';
+
+export interface IQueueState {
+  dataTaskFinished: boolean;
+}
+
+const initialState: IQueueState = {
+  dataTaskFinished: null,
+};
+
+const queueReducer: Reducer<IQueueState> = (
+  state: IQueueState = initialState,
+  action?: Action,
+): IQueueState => {
+  if (!action) {
+    return state;
+  }
+  switch (action.type) {
+    case DataTaskQueueActionTypes.START_INITIAL_DATA_TASK_QUEUE: {
+      return update<IQueueState>(state, {
+        dataTaskFinished: {
+          $set: false,
+        },
+      });
+    }
+    // TODO: START_INITIAL_DATA_TASK_QUEUE_FULFILLED
```

Comment:
This will never update `dataTaskFinished` to true?

Reply:
Added fulfilled.